overlord/ifacestate/ifacestate.go

```diff
@@ -170,8 +170,8 @@ func Disconnect(st *state.State, plugSnap, plugName, slotSnap, slotName string)
 
 // CheckInterfaces checks whether plugs and slots of snap are allowed for installation.
 func CheckInterfaces(st *state.State, snapInfo *snap.Info) error {
-	// XXX: AddImplicitSlots is really a brittle interface
-	snap.AddImplicitSlots(snapInfo)
+	// XXX: addImplicitSlots is really a brittle interface
+	addImplicitSlots(snapInfo)
```

Comment:
Yes but this is not something I wish to do in this branch. The goal right now is to reduce the number of files that need to be patched to create a new interface. Addressing the implicit slot mechanism can and should be done separately IMO. I think this branch does improve the situation slightly as it moves the code closer to the interface manager where ultimately all the interface decisions are being taken.